### ISSUE TITLE
Discourse plugins: "ELOOP: Too many levels of symbolic links "

### ISSUE THREAD

**Issue by ryantm** [NixOS/nixpkgs#119584]
**Describe the bug**

When using some plugins with the new Discourse Module, it creates a directory structure with too many symbolic links.

**To Reproduce**
My configuration looks like:
```
{
    systemd.tmpfiles.rules = [
      "d  /etc/discourse 0755 discourse discourse - -"
    ];

    networking.firewall.allowedTCPPorts = [ 80 443 ];

    services.discourse = {
      enable = true;
      hostname = "our-discourse.example.com";
      admin = {
        email = "discourse@example.com";
        username = "admin";
        fullName = "Administrator";
        passwordFile = config.deployment.keys.discourse-admin-password.path;
      };
      mail.outgoing = {
        serverAddress = "email-service";
        port = 587;
        username = "email-username";
        passwordFile  = config.deployment.keys.discourse-email-password.path;
      };
      plugins = [
        (pkgs.fetchFromGitHub {
          owner = "discourse";
          repo = "discourse-solved";
          rev = "704d2df27a61ac8f9c9e7c0851c1271d45df393d";
          sha256 = "sha256:0r3w9igv9y2acqlxzw1lb6x70ara0ab09v0ivcbkl06skdigf4l6";
        })
        (pkgs.fetchFromGitHub {
          owner = "discourse";
          repo = "discourse-math";
          rev = "143ddea4558ea9a1b3fd71635bc11e055763c8e7";
          sha256 = "sha256:18pq5ybl3g34i39cpixc3nszvq8gx5yji58zlbbl6428mm011cbx";
        })
      ];
      sslCertificate = config.our.certChainPath;
      sslCertificateKey = config.our.privateKeyPath;
    };
}
```
**Expected behavior**
Discourse runs

**Additional context**
```
Apr 15 16:08:37 host systemd[1]: Starting discourse.service...
Apr 15 16:08:41 host discourse-pre-start[11490]: rake aborted!
Apr 15 16:08:41 host discourse-pre-start[11490]: Errno::ELOOP: Too many levels of symbolic links - /nix/store/rpmy4093wvzy3vg0cccrh3mnaia6hp0r-discourse-2.6.3/share/discourse/plugins/ln/config/locales
Apr 15 16:08:41 host discourse-pre-start[11490]: /nix/store/rpmy4093wvzy3vg0cccrh3mnaia6hp0r-discourse-2.6.3/share/discourse/config/application.rb:208:in `[]'
Apr 15 16:08:41 host discourse-pre-start[11490]: /nix/store/rpmy4093wvzy3vg0cccrh3mnaia6hp0r-discourse-2.6.3/share/discourse/config/application.rb:208:in `<class:Application>'
Apr 15 16:08:41 host discourse-pre-start[11490]: /nix/store/rpmy4093wvzy3vg0cccrh3mnaia6hp0r-discourse-2.6.3/share/discourse/config/application.rb:71:in `<module:Discourse>'
Apr 15 16:08:41 host discourse-pre-start[11490]: /nix/store/rpmy4093wvzy3vg0cccrh3mnaia6hp0r-discourse-2.6.3/share/discourse/config/application.rb:70:in `<top (required)>'
Apr 15 16:08:41 host discourse-pre-start[11490]: /nix/store/rpmy4093wvzy3vg0cccrh3mnaia6hp0r-discourse-2.6.3/share/discourse/Rakefile:7:in `require'
Apr 15 16:08:41 host discourse-pre-start[11490]: /nix/store/rpmy4093wvzy3vg0cccrh3mnaia6hp0r-discourse-2.6.3/share/discourse/Rakefile:7:in `<top (required)>'
Apr 15 16:08:41 host discourse-pre-start[11490]: /nix/store/k8lda3816mqp8ijhi31awpg12mp3zi78-discourse-ruby-env-2.6.3/lib/ruby/gems/2.7.0/gems/rake-13.0.1/exe/rake:27:in `<top (required)>'
Apr 15 16:08:41 host discourse-pre-start[11490]: (See full trace by running task with --trace)
```

```
# ls -la /nix/store/rpmy4093wvzy3vg0cccrh3mnaia6hp0r-discourse-2.6.3/share/discourse/plugins/ln/
ls: cannot access '/nix/store/rpmy4093wvzy3vg0cccrh3mnaia6hp0r-discourse-2.6.3/share/discourse/plugins/ln/': Too many levels of symbolic links
```

with no plugins specified it runs Discourse fine.

**Notify maintainers**

@talyz 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.13, NixOS, 21.05pre-git (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - nixpkgs: `/run/current-system/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: discourse
```

**Comment by ryantm** [comment]
There is a missing newline in the Discourse prestart script

```
cp -r /nix/store/rpmy4093wvzy3vg0cccrh3mnaia6hp0r-discourse-2.6.3/share/discourse/plugins.dist/* /run/discourse/plugins/
ln -sf /nix/store/1faxwgpx6ax99a0idnwngajlf0wjzc40-source /run/discourse/plugins/ln -sf /nix/store/021mx28rzsw72ddp4abpl1fq4lrglzqc-source /run/discourse/plug
```

https://github.com/NixOS/nixpkgs/blob/936f450e3d80d9fe42ad4d825313e766e40919a1/nixos/modules/services/web-apps/discourse.nix#L705

**Comment by talyz** [comment]
Oh, oops, it should be fixed in #119596.

**Comment by ryantm** [comment]
Thanks! I confirmed this is fixed by the changes.
